instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -98,6 +125,31 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
         val refreshedRoutes = initialRouteIdlingResource.next()
         initialRouteIdlingResource.unregister()
 
+        val progressIdlingResource = RouteProgressStateIdlingResource(
+            mapboxNavigation,
+            RouteProgressState.TRACKING
+        )
+        progressIdlingResource.register()
+        Espresso.onIdle()
+        progressIdlingResource.unregister()
+
+        val latch = CountDownLatch(1)
+        runOnMainSync {
+            mapboxNavigation.registerRouteProgressObserver { routeProgress ->
+                val durationRemaining = routeProgress.durationRemaining
+                val expectedDurationRemaining = 1180.651
+                // 30 seconds margin of error
```

Comment:
why is it 30 seconds? Should it be the exact value as we use mockk data?

Reply:
I'm not sure. This is an integration test after all, I assume native navigator could introduce a margin of error depending on how the location we provide is snapped to the route.